validator/sawtooth_validator/execution/executor.py

```diff
@@ -77,47 +102,188 @@ def run(self):
                 header.family_version,
                 header.payload_encoding)
 
-            # Currently we only check for the sawtooth_config txn family,
-            # as it is the only family we know to require.
-            if self._require_txn_processors and \
-                    header.family_name == 'sawtooth_config' and \
-                    processor_type not in self._processors:
-                # wait until required processor is registered:
-                LOGGER.info('Waiting for transaction processor (%s, %s, %s)',
-                            header.family_name,
-                            header.family_version,
-                            header.payload_encoding)
-
-                self._processors.wait_to_process(processor_type)
-
-            if processor_type not in self._processors:
-                raise Exception("internal error, no processor available")
-            processor = self._processors.get_next_of_type(processor_type)
+            config = self._config_view_factory.create_config_view(
+                txn_info.state_hash)
+            transaction_families = config.get_setting(
+                key=self._tp_config_key,
+                default_value="[]")
+            # After reading the transaction families required in configuration
+            # try to json.loads them into a python object
+            # If there is a misconfiguration, proceed as if there is no
+            # configuration.
+            try:
+                transaction_families = json.loads(transaction_families)
+                required_transaction_processors = [
+                    processor_iterator.ProcessorType(
+                        d.get('family'),
+                        d.get('version'),
+                        d.get('encoding')) for d in transaction_families]
+            except ValueError:
+                LOGGER.warning("sawtooth.validator.transaction_families "
+                               "misconfigured. Expecting a json array, found"
+                               " %s", transaction_families)
+                required_transaction_processors = []
+
+            config_processor = processor_iterator.ProcessorType(
+                name='sawtooth_config',
+                version='1.0',
+                encoding='application/protobuf')
+
+            # If there are required transaction processors, sawtooth_config
```

Comment:
Since config should be explicitly set, it seems like there needs to be a check that there is some transaction processor with the capabilities to process config transactions. The validator is misconfigured otherwise. I agree that checking for a specific name, version, encoding isn't right, when for the lifetime of a chain there could be different config processors. Could we require that any config processor has "config" in the name? I think we are inviting bad situations where, after a mistake in changing required transaction processors, we fail every config transaction and aren't able to make another config change.